apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DecimalReadTests.cs

```diff
@@ -0,0 +1,211 @@
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Spanner.V1;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.IntegrationTests
+{
+    [Collection(nameof(DecimalTableFixture))]
+    public class DecimalReadTests
+    {
+        private readonly DecimalTableFixture _fixture;
+
+        public DecimalReadTests(DecimalTableFixture fixture) =>
+            _fixture = fixture;
+
+        public static IEnumerable<object[]> TestDecimals =>
+            new List<object[]>
+            {
+                new object[] { new decimal(1.23456789) },
+                new object[] { decimal.MinusOne },
+                new object[] { decimal.Zero },
+                new object[] { new decimal(123456789.0123456789012345678) },
+            };
+
+        public static IEnumerable<object[]> TestDecimalsWithNumeric => 
+            Enumerable.Concat(TestDecimals,
+            new List<object[]>
+            {
+                new object[] { decimal.MaxValue },
+                new object[] { decimal.MinValue },             
+            });
+
+        // Test case: Simple case of using decimal value with Float64 column in database.
+        // SpannerDbType is explicitly set for the parameter.
+        [Theory]
+        [MemberData(nameof(TestDecimals))]
+        public async Task WriteThenRead_ShouldBeEqual(decimal expectedValue)
+        {
+            using var connection = _fixture.GetConnection();
+            // Write the decimal value and read it back.
+            await connection.CreateInsertOrUpdateCommand(_fixture.TableName,
+                new SpannerParameterCollection
+                {
+                    new SpannerParameter("DecimalValue", SpannerDbType.Float64, expectedValue),
+                }
+            ).ExecuteNonQueryAsync();
+            var decimalValue = await connection.CreateSelectCommand($"SELECT DecimalValue FROM {_fixture.TableName}").ExecuteScalarAsync<decimal>();
+            Assert.Equal(expectedValue, decimalValue);
+        }
+
+        // Test Case: Using decimal with Float64 column in database. SpannerConversionOptions set to use Numeric for decimal.
+        // SpannerDbType is explicitly set for the parameter.
+        // The configuration should be ignored in .NET to Spanner path, as SpannerDbType of parameter is specified.
+        // The configuration should be honored in Spanner to .NET path if ClrType isn't specified while reading.
+        [Theory]
+        [MemberData(nameof(TestDecimals))]
+        public async Task WriteThenRead_ShouldBeEqual_UseOptions(decimal expectedValue)
```

Comment:
If this is meant to be for *read* tests, I'd expect to just insert rows into the table to start with.
The parameter type used to insert the data won't change how the data from that table is read.

It would be fine to have tests that use a parameter and basically return that parameter value, without reading from any actual table. (I can't remember the syntax for that right now, but it's there somewhere.)

Basically I think we need a bit more clarity about what each test is meant to achieve. I see it as:

- When writing, how do we send a parameter to Spanner
- When reading, how do we receive a value from Spanner

... in each case of "explicitly use Float64, explicitly use Numeric, implicitly use Float64, implicitly use Numeric due to connection options".